program/src/processor.rs

```diff
@@ -793,6 +848,10 @@ pub fn process_withdraw(
         return Err(LidoError::ValidatorWithMoreStakeExists.into());
     }
 
+    // Nit: this also creates the same kind of confusion as the unstake instruction,
+    // where it only uses `stake_seeds.begin`, so if there are multiple stake
+    // accounts, we have to merge before performing a withdraw, which may
+    // be confusing.
```

Comment:
We only allow withdrawing from the first stake account, this is on purpose. The maintainer bot merges as many stake accounts as it can, so in the best case, we have one that’s fully active, and one that’s activating, that we can’t withdraw from anyway.

There was a bug checking which validator to withdraw from, which we fixed in https://github.com/ChorusOne/solido/pull/391.